moved/src/move_execution/evm_native.rs

```diff
@@ -0,0 +1,862 @@
+            return exists_in_prior_changes;
+        }
+        // If not in the prior changes then check the resolver
+        resolver
+            .get_resource(&EVM_NATIVE_ADDRESS, struct_tag)
+            .map(|x| x.is_some())
+            .unwrap_or(false)
+    };
+
+    // Push AccountInfo resource
+    let struct_tag = account_info_struct_tag(address);
+    let account_info = account_info_to_move_value(&account.info, code_hash)
+        .simple_serialize(&ACCOUNT_INFO_LAYOUT)
+        .expect("Account info must serialize");
+    let is_created = !resource_exists(&struct_tag);
+    let op = if is_created {
+        Op::New(account_info.into())
+    } else {
+        Op::Modify(account_info.into())
+    };
+    result
+        .add_resource_op(struct_tag, op)
+        .expect("Resource cannot already exist in result");
+
+    // Push CodeHash resource if needed.
+    // We don't need to push anything if the resource already exists.
+    let struct_tag = code_hash_struct_tag(&code_hash);
+    let code_resource_exists = resource_exists(&struct_tag);
+    if !code_resource_exists {
+        if let Some(code) = &account.info.code {
+            if !code.is_empty() {
+                let struct_tag = code_hash_struct_tag(&code_hash);
+                let code = Value::vector_u8(code.original_bytes())
+                    .simple_serialize(&CODE_LAYOUT)
+                    .expect("EVM code must serialize");
+                let op = Op::New(code.into());
+                // If the same contract is deployed more than once then the same resource
+                // could be added twice, but that's ok we can just skip the duplicate.
+                result.add_resource_op(struct_tag, op).ok();
+            }
+        }
+    }
+
+    // TODO: If an address self-destructs and then is re-created then its storage
+    // must be entirely reset. With the current model we cannot easily delete all the
+    // storage for an account (we would need to loop through all the resources for
+    // the EVM native). Therefore this may need a redesign if we decide to support
+    // EVM self-destruct.
+    for (index, value) in account.changed_storage_slots() {
+        let struct_tag = account_storage_struct_tag(address, index);
+        let op = if value.present_value.is_zero() {
+            Op::Delete
+        } else {
+            let move_value = Value::u256(to_move_u256(&value.present_value))
+                .simple_serialize(&ACCOUNT_STORAGE_LAYOUT)
+                .expect("EVM storage value must serialize");
+            if value.original_value.is_zero() {
+                Op::New(move_value.into())
+            } else {
+                Op::Modify(move_value.into())
+            }
+        };
+        result
+            .add_resource_op(struct_tag, op)
+            .expect("Cannot have duplicate storage index");
+    }
+}
+
+fn account_info_struct_tag(address: &Address) -> StructTag {
+    let name = format!("Info_{}", address.encode_hex());
```

Comment:
We might wanna call this `Account_{}` instead 🤔

Reply:
Changed in https://github.com/MovedNetwork/op-move/pull/143/commits/9cdd48aab42f21335a3b1a48083a716b7efb14fe